File: js/01_mar/zipArray/zipArray.java
import static java.lang.System.out;
import java.util.HashMap;

class zipArray {
	private static String[] firstArr = {"abc", "3", "yo"};
	private static String[] secondArr = {"42", "wassup", "true"};

	public static void main(String[] args) {
		out.println(zipArr(firstArr, secondArr));
	}

	private static HashMap<String, String> zipArr(String[] arr1, String[] arr2) {
		HashMap<String, String> resMap = new HashMap<String, String>();

		int n = arr1.length > arr2.length ? arr2.length : arr1.length;

		for (int i = 0; i < n; i++) {
			resMap.put(arr1[i], arr2[i]);
		}

		return resMap;
	}
}

File: js/01_mar/zipArray/zipArray.js
const zipArr = (arr1, arr2) => {
	let resObj = {},
		n = arr1.length > arr2.length ? arr2.length : arr1.length;

	for (let i = 0; i < n; i++) {
		resObj[arr1[i]] = arr2[i];
	}

	return resObj;
};

const firstArr = ['abc', 3, 'yo'],
	secondArr = [42, 'wassup', true];

console.log(zipArr(firstArr, secondArr));
